Доработана функциональность по техническому заданию спринта 8

Dmitry92111 opened the pull request:
- добавлены новые характеристики задач (время начала, длительность, время окончания)
- добавлена возможность сортировки и получения задач в порядке приоритета (от самой срочной)
- добавлена проверка на то что задачи не пересекутся по времени
- циклы forEach заменены на stream API
- написаны тесты для новой функциональности

Dmitry92111 committed:
Доработана функциональность по техническому заданию спринта 8

- добавлены новые характеристики задач (время начала, длительность, время окончания)
- добавлена возможность сортировки и получения задач в порядке приоритета (от самой срочной)
- добавлена проверка на то что задачи не пересекутся по времени
- циклы forEach заменены на stream API
- написаны тесты для новой функциональности

Zurus commented on src/ru/common/model/EpicTask.java:
```diff
@@ -26,4 +40,8 @@ public ArrayList<Integer> getRelatedSubTasksId() {
     public void removeRelatedSubTaskId(int subTaskId) {
         relatedSubTasksId.remove((Integer) subTaskId);
     }
+
```
В EpicTask метод getEndTime будет возвращать реализацию из класса Task.
Что не совсем корректно. Предлагаю переопределить этот метод.

Zurus commented on src/ru/common/managers/InMemoryTaskManager.java:
```diff
@@ -38,23 +48,34 @@ public void addNewEpicTask(EpicTask epicTask) {
-        if (task.getClass() != Task.class) {
-            return;
-        }
+        if (task.getClass() != Task.class) return;
```
Предлагаю использовать метод getType.

Так же хотелось бы заменить в проекте все использования instanceOf на getType.

Zurus commented on src/ru/common/managers/InMemoryTaskManager.java:
```diff
@@ -1,31 +1,41 @@
     public void addNewTask(Task task) {
-        if (task.getClass() != Task.class) {
-            return;
+        if (task.getClass() != Task.class) return;
```
Предлагаю заменить на getType.

Zurus commented on test/FileBackedTaskManagerTest.java:
```diff
@@ -1,36 +1,31 @@
-        task3 = new Task("task 3", "taskDescription 3");
-        epicTask1 = new EpicTask("epicTask 1", "epicTaskDescription 1");
-        epicTask2 = new EpicTask("epicTask 2", "epicTaskDescription 2");
+        createTasks();
```
Предлагаю этот момент немного доработать.

Сейчас мы создаем в тестовых данных только Task/Epic. 
Это не совсем хорошо, так как ограничивает классы которые мы можем протестировать.
Предложение:
1) Вызывать createTasks() для каждого конкретного теста.
2) Добавить разные наборы моковых данных. task/epic/subtask.

Dmitry92111 committed:
- заменил все instanseOf и getClass по всему коду на get.Type для получения типов задач

- в целях тестировая создал отдельные методы для создания task/epic/subtask и метод для всего сета

- переопределил метод получения endTime в классе EpicTask
